merge/real_test.go

```diff
@@ -105,44 +106,44 @@ func BenchmarkOperations(b *testing.B) {
 					ops: []Operation{
 						Update{
 							Manager:    "controller",
-							APIVersion: "v1",
+							APIVersion: fieldpath.APIVersion(bench.typename),
 							Object:     bench.obj,
 						},
 						Update{
 							Manager:    "other-controller",
-							APIVersion: "v1",
-							Object:     bench.obj,
-						},
-					},
-				},
-				{
-					name: "UpdateVersion",
-					ops: []Operation{
-						Update{
-							Manager:    "controller",
-							APIVersion: "v1",
-							Object:     bench.obj,
-						},
-						Update{
-							Manager:    "other-controller",
-							APIVersion: "v2",
+							APIVersion: fieldpath.APIVersion(bench.typename),
 							Object:     bench.obj,
 						},
 					},
 				},
+				// XXX: How do we do that?
```

Comment:
I need to think about that a little bit

Reply:
This test doesn't have a lot of value, maybe we should just drop it ...

Reply:
This condition should be tested, is it tested elsewhere?

We'd expect the managed fields to no longer reference v1 (assuming all fields are in v2), right? That seems useful to check somehow.

Reply:
Yes, but we don't have to do that with these objects because it's much harder here.

Reply:
Can you just replace `tc.Bench(parser)` below, with something like `tc.Bench(SameVersionParser{T: parser.Type(bench.typename)})` (I think you would also have to wire typename through tc since this is in a different loop)

Reply:
I think that can work yeah ... thanks!